src/main/java/gregtech/client/utils/RenderBufferHelper.java

```diff
@@ -47,6 +48,10 @@ public static void renderCubeFrame(BufferBuilder buffer, double minX, double min
         buffer.pos(minX, maxY, maxZ).color(r, g, b, a).endVertex();
     }
 
+    public static void renderCubeFace(BufferBuilder buffer, Cuboid6 cuboid, float r, float g, float b, float a) {
+        renderCubeFace(buffer, cuboid.min.x, cuboid.min.y, cuboid.min.z, cuboid.max.x, cuboid.max.y, cuboid.max.z, r, g, b, a);
+    }
+
```

Comment:
because `renderCubeFace` always render all faces with the same color, making the cube looks no edges. could add a parameter `boolean shading`. Do shading as what minecraft does that each faces have a little different wight. then render faces that multiply the weight by the color. dunno if you want to do it, it might not make much difference in result, but can see edges.

Reply:
http://greyminecraftcoder.blogspot.com/2020/04/lighting-1144.html